### ISSUE TITLE
QNAP snmpwalk

### ISSUE THREAD

**Issue by Wolvverine** [fusioninventory/sysobject.ids#24]
[snmpwalk_qnap.txt](https://github.com/fusioninventory/sysobject.ids/files/2428057/snmpwalk_qnap.txt)

**Comment by g-bougard** [comment]
Hi @Wolvverine 
your walk does not include the `SNMPv2-MIB::sysObjectID` or `.1.3.6.1.2.1.1.2` value.
Did you follow the [snmpwalk output documentation](http://fusioninventory.org/documentation/bugreport/#snmpwalk-output) ?
Maybe your walk is not complete.

**Comment by Wolvverine** [comment]
new upload

**Comment by g-bougard** [comment]
Humpf,
it seems your walk start from `.1.3.6.1.2.1.1.9.1.2.1` ... some lines should has been removed at the beginning.

**Comment by g-bougard** [comment]
Hi @Wolvverine 
about my last comment, I mean `SNMPv2-MIB::sysObjectID` or `.1.3.6.1.2.1.1.2` value is still missing as first lines of your walk seems to have not been included. The MIB won't make that lines appear by themselves.

**Comment by Wolvverine** [comment]
Maybe because I'm walk in v3 ?
.1.3.6.1.4.1.8072.1.2.1.1.4.0.8.1.3.6.1.2.1.1.2.127 = STRING: mibII/sysObjectID

**Comment by g-bougard** [comment]
@Wolvverine 
I don't think so. Maybe you truncated some lines when updating the walk. Can you double-check and attach it in a new comment ?

**Comment by Wolvverine** [comment]
![walk](https://user-images.githubusercontent.com/5873929/46289834-755d2f00-c58a-11e8-85b4-8ac8cd0d0322.JPG)

snmpalk_command  | wc -l
3765

**Comment by g-bougard** [comment]
The file you provided still have 3799 lines... which is larger than 3765.
Btw I guess the user you're using may not have a full access on the device OIDs.

What are you get while doing something like:
`snmpget -v2c -c public "serverIP" 1.3.6.1.2.1.1.2.0`
but changing the authentication by your snmp v3 authentication ?

**Comment by Wolvverine** [comment]
without .1
.1.3.6.1.2.1.1.1.0 = STRING: Linux TS-221 4.3.3

**Comment by Wolvverine** [comment]
@g-bougard  some lines with hex value are split to 2 lines in editor

**Comment by g-bougard** [comment]
Do you mean you have not result with `1.3.6.1.2.1.1.2.0` ?
If you don't have any result for this OID, you may have an issue with your user or your device can't be discovered by the agent (but this would be the first time I hear about a SNMP device not providing this OID).

**Comment by Wolvverine** [comment]
result with 1.3.6.1.2.1.1.2.0
.1.3.6.1.2.1.1.2.0 = OID: .1.3.6.1.4.1.24681

**Comment by g-bougard** [comment]
Okay, can you complete with this ones:
```
1.3.6.1.2.1.1.3.0
1.3.6.1.2.1.1.4.0
1.3.6.1.2.1.1.5.0
1.3.6.1.2.1.1.6.0
```
This should be sufficient do complete you walk manually.
Thank you

**Comment by Wolvverine** [comment]
.1.3.6.1.2.1.1.3.0 = Timeticks: (42826661) 4 days, 22:57:46.61
.1.3.6.1.2.1.1.4.0 = STRING:
.1.3.6.1.2.1.1.5.0 = STRING: QNAP
.1.3.6.1.2.1.1.6.0 = STRING:

**Comment by g-bougard** [comment]
Hi @Wolvverine 
in fact, it seems SNMP with QNAP devices makes a problem to be inventoried if we can't find the hostname by default from `.1.3.6.1.2.1.1.5.0`. I proposed a [PR #567 on agent](https://github.com/fusioninventory/fusioninventory-agent/pull/567) to set a default hostname when we find a serial number (the case with your walk). You still can try the patch from this PR to see if it fixes your case.

After that, I updated the sysobject.ids in cfafee48b2 to include QNAP support, but QNAP was still supported as the description matches the `Linux TS-` pattern. In debug mode, you'll obtain a warning telling sysobjetcID is invalid as we only find the manufacturer id, but this is followed with partial match and is sufficient.

Then I'm closing this issue here. Feel free to open a [FusionInventory-Agent issue](https://github.com/fusioninventory/fusioninventory-agent/issues) linking this one if nor the PR nor the last commit here helps to inventory your QNAP devices.
